pyproject.toml

```diff
@@ -15,12 +15,50 @@ log_cli_level = "INFO"
 # Formatting tools configuration
 [tool.black]
 line-length = 99
-target-version = ["py38"]
+target-version = ["py312"]
 
 [tool.isort]
 line_length = 99
 profile = "black"
 
+[tool.poetry]
+name = "kubeflow-uat"
+version = "1.0"
+description = "kubeflow-uat"
+authors = []
+
+[tool.poetry.dependencies]
+python = ">=3.10"
+juju = "<4.0"
+lightkube = ">=0.15"
+# pytest is transitive dependencies of other packages
+# pytest = "^7.4"
+pytest-operator = ">=0.31"
```

Comment:
Why are we pinning all those dependencies ... we had a practice to do that only if it is necessary .... where are those versions comming from ?

Reply:
yes, I believe it is sensible. I have reverted to fully open requirements and updated locked environment